### ISSUE TITLE
adapt_framework: headings lose focus if completed with focus

### ISSUE THREAD

**Issue by oliverfoster** [adaptlearning/adapt_framework#3132]
### Subject of the issue
As the heading view performs a destructive rerender the focus goes missing if the heading has focus when it is rerendered (when a component completes).
